### PR TITLE
Transition to monteway

### PR DESCRIPTION
**Pull request by RubyDiver**

### CODE REVIEW

File: app/controllers/api/v1/cinema_halls_controller.rb [hunk 1]
```diff
@@ -7,16 +7,16 @@ def index
   end
 
   def show
-    @cinema_hall = CinemaHalls::UseCases::Find.new.call(id: params[:id])
+    @cinema_hall = CinemaHalls::UseCases::Show.new.call(id: params[:id])
```
Comment (baniak66) [inline app/controllers/api/v1/cinema_halls_controller.rb]: You don't need that use_case, you can use repository class: `CinemaHalls::Repository.new.find(params[:id])`

File: app/controllers/api/v1/seances_controller.rb [hunk 2]
```diff
@@ -1,45 +1,43 @@
 class Api::V1::SeancesController < ApplicationController
-  before_action :set_seance, only: [:destroy, :update, :show]
 
   def index
-    @seances = Seance.all
-    render json: @seances
+    @seances = Seances::UseCases::Index.new.call
```
Comment (baniak66) [inline app/controllers/api/v1/seances_controller.rb]: Same here: `Seances::Repository.new.find_all`

File: app/controllers/api/v1/seances_controller.rb [hunk 3]
```diff
@@ -1,45 +1,43 @@
...
+    render json: Seances::Representers::AllSeances.new(@seances).basic
   end
 
   def show
-    render json: @seance
+    @seance = Seances::UseCases::Show.new.call(id: params[:id])
```
Comment (baniak66) [inline app/controllers/api/v1/seances_controller.rb]: `Seances::Repository.new.find(params[:id])`

**Review by baniak66 — COMMENTED** [review]
Overall LGTM, just small remarks 👍

File: app/controllers/api/v1/cinema_halls_controller.rb (hunk 1 above)
Comment (RubyDiver) [reply app/controllers/api/v1/cinema_halls_controller.rb]: So, I don't need use_case index? at all?

Comment (baniak66) [reply app/controllers/api/v1/cinema_halls_controller.rb]: That's right. In fact `CinemaHalls::UseCases::Show.new.call(id: params[:id])` just call `CinemaHalls::Repository.new.find(params[:id])`, nothing more. The same with Index use_case.

Comment (RubyDiver) [reply app/controllers/api/v1/cinema_halls_controller.rb]: Now the problem is that POST test seance and POST test reservation say 422.

File: app/controllers/api/v1/seances_controller.rb [hunk 4]
```diff
@@ -6,14 +6,14 @@ def index
...
+    @seance = Seances::Repository.new.find(params[:id])
     render json: Seances::Representers::OneSeance.new(@seance).basic
   end
 
   def create
-    @cinema_hall = CinemaHall.find(params[:cinema_hall_id])
-    @seance = @cinema_hall.seances.create(seance_params)
-    @seance = Seances::UseCases::Create.new.call(params: seance_params)
+    @cinema_hall = CinemaHalls::UseCases::Show.new.call(id: params[:cinema_hall_id])
+    @seance = Seances::UseCases::Create.new.call(params: seance_params.merge(cinema_hall_id: params[:cinema_hall_id]))
```
Comment (RubyDiver) [inline app/controllers/api/v1/seances_controller.rb]: @seance = Seances::UseCases::Create.new.call(params: seance_params.merge(cinema_hall_id: params[:cinema_hall_id]))
  @seance = Seances::UseCases::Create.new.call(params: seance_params.merge(movie_id: params[:movie_id]))
 I need to connect those 2 create method into 1, and I don't know how. @w0st

Comment (w0st) [reply app/controllers/api/v1/seances_controller.rb]: Let's discuss it during the call, we can simplify it.

**Pull request merged**